incubator/binding-asyncapi/src/main/java/io/aklivity/zilla/runtime/binding/asyncapi/internal/stream/AsyncapiClientFactory.java

```diff
@@ -0,0 +1,1064 @@
+            final long acknowledge = end.acknowledge();
+            final long traceId = end.traceId();
+            final OctetsFW extension = end.extension();
+
+            assert acknowledge <= sequence;
+            assert sequence >= replySeq;
+
+            replySeq = sequence;
+            state = AsyncapiState.closingReply(state);
+
+            assert replyAck <= replySeq;
+
+            delegate.doMqttEnd(traceId, extension);
+        }
+
+        private void onMqttAbort(
+            AbortFW abort)
+        {
+            final long sequence = abort.sequence();
+            final long acknowledge = abort.acknowledge();
+            final long traceId = abort.traceId();
+
+            assert acknowledge <= sequence;
+            assert sequence >= replySeq;
+
+            replySeq = sequence;
+            state = AsyncapiState.closingReply(state);
+
+            assert replyAck <= replySeq;
+
+            delegate.doMqttAbort(traceId);
+        }
+
+        private void doMqttReset(
+            long traceId)
+        {
+            if (!AsyncapiState.replyClosed(state))
+            {
+                doReset(receiver, originId, routedId, replyId, replySeq, replyAck, replyMax,
+                    traceId, authorization, EMPTY_OCTETS);
+
+                state = AsyncapiState.closeReply(state);
+            }
+        }
+
+        private void doMqttWindow(
+            long traceId,
+            long authorization,
+            long budgetId,
+            int padding)
+        {
+            replyAck = Math.max(delegate.replyAck - replyPad, 0);
+            replyMax = delegate.replyMax;
+
+            doWindow(receiver, originId, routedId, replyId, replySeq, replyAck, replyMax,
+                traceId, authorization, budgetId, padding + replyPad);
+        }
+    }
+
+    private final class AsyncapiStream
+    {
+        private final MqttStream mqtt;
+        private final MessageConsumer sender;
+        private final String operationId;
+        private final long originId;
+        private final long routedId;
+        private final long initialId;
+        private final long replyId;
+        private final long affinity;
+        private final long authorization;
+
+        private int state;
+
+        private long replyBudgetId;
+
+        private long initialSeq;
+        private long initialAck;
+        private int initialMax;
+
+        private long replySeq;
+        private long replyAck;
+        private int replyMax;
+        private int replyPad;
+        private long replyBud;
+        private int replyCap;
+
+        AsyncapiStream(
+            MessageConsumer sender,
+            long originId,
+            long routedId,
+            long initialId,
+            long affinity,
+            long authorization,
+            long resolvedId,
+            String operationId)
+        {
+            this.mqtt =  new MqttStream(this, routedId, resolvedId, authorization);
```

Comment:
Does the behavior of this stream change if it targets `http` instead of `mqtt` in a composite namespace?
If not, then perhaps rename to `CompositeStream`?